litebird_sim/spherical_harmonics.py

```diff
@@ -221,3 +316,213 @@ def resize_alm(
+
+    def __rmul__(self, other: Union[float, np.ndarray]):
+        return self.__mul__(other)
+
+    def convolve(
+        self, f_ell: Union[np.ndarray, list[np.ndarray]]
+    ) -> "SphericalHarmonics":
+        """
+        Apply a beam or filter to the SH coefficients.
+
+        Parameters
+        ----------
+        f_ell : np.ndarray or list[np.ndarray]
+            The ℓ-dependent filter(s). Must be of shape (lmax+1,) or (nstokes, lmax+1)
+
+        Returns
+        -------
+        SphericalHarmonics
+            A new SphericalHarmonics object with filtered coefficients.
+        """
+        l_arr = self.alm_l_array(self.lmax, self.mmax)
+
+        if isinstance(f_ell, np.ndarray):
+            if f_ell.ndim == 1:
+                kernel = np.broadcast_to(f_ell[l_arr], self.values.shape)
+            elif f_ell.ndim == 2 and f_ell.shape[0] == self.nstokes:
+                kernel = np.stack([f[l_arr] for f in f_ell])
+            else:
+                raise ValueError(f"Invalid shape for f_ell: {f_ell.shape}")
+        elif isinstance(f_ell, list):
+            if len(f_ell) != self.nstokes:
+                raise ValueError(f"Expected {self.nstokes} filters, got {len(f_ell)}")
+            kernel = np.stack([np.asarray(f)[l_arr] for f in f_ell])
+        else:
+            raise TypeError("f_ell must be a numpy array or a list of numpy arrays")
+
+        return SphericalHarmonics(
+            values=self.values * kernel, lmax=self.lmax, mmax=self.mmax
+        )
+
+    def copy(self):
+        """Returns a deep copy of this SphericalHarmonics object."""
+        return SphericalHarmonics(
+            values=self.values.copy(), lmax=self.lmax, mmax=self.mmax
+        )
+
+    def __sub__(self, other):
+        if not isinstance(other, SphericalHarmonics):
+            raise TypeError("Subtraction requires another SphericalHarmonics object")
+        if (
+            self.lmax != other.lmax
+            or self.mmax != other.mmax
+            or self.nstokes != other.nstokes
+        ):
+            raise ValueError(
+                "SphericalHarmonics objects must have matching lmax, mmax, and nstokes"
+            )
+        return SphericalHarmonics(
+            values=self.values - other.values, lmax=self.lmax, mmax=self.mmax
+        )
+
+    def __isub__(self, other):
+        if not isinstance(other, SphericalHarmonics):
+            raise TypeError("Subtraction requires another SphericalHarmonics object")
+        if (
+            self.lmax != other.lmax
+            or self.mmax != other.mmax
+            or self.nstokes != other.nstokes
+        ):
+            raise ValueError(
+                "SphericalHarmonics objects must have matching lmax, mmax, and nstokes"
+            )
+        self.values -= other.values
+        return self
+
+    def __eq__(self, other):
+        if not isinstance(other, SphericalHarmonics):
+            return False
```

Comment:
This condition means that the user is comparing apples with oranges, which is probably not what they want. Perhaps it would be better to raise a `ValueError` exception, stating that only `SphericalHarmonics` objects can be compared.